presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -30,16 +31,19 @@
 import java.util.Locale;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Objects;
 import java.util.Optional;
 import java.util.TimeZone;
 
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
 public final class Session
 {
     private final QueryId queryId;
+    private final Optional<TransactionId> transactionId;
```

Comment:
Is it possible to make this required? It looks like there's only a few places that deal with auto-commit, where it's important (and maybe that could be a separate property?). Everything looks like it just fails if the transaction id is missing.
I generally dislike `Optional` because I feel like it's only a minor improvement over `@Nullable`